EsoTech.MessageQueue/AzureServiceBus/AzureServiceBusManager.cs

```diff
@@ -0,0 +1,148 @@
+                    {
+                        Name = ruleName,
+                        Filter = new CorrelationFilter
+                        {
+                            Properties = { ["EsoTechMessageKind"] = messageType }
+                        }
+                    });
+
+                subscriptions.Add(subscriptionName);
+            }
+
+            stopwatch.Stop();
+            _logger.LogInformation($"Subscription {subscriptionName} updated for {stopwatch.ElapsedMilliseconds}ms");
+        }
+
+        public async Task UpdateTopic(string topicName)
+        {
+            _logger.LogInformation($"Updating topick {topicName}");
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+
+            if (!_topicks.ContainsKey(topicName))
+            {
+                if (!await Client.TopicExistsAsync(topicName))
+                    await Client.CreateTopicAsync(new TopicDescription(topicName)
+                    {
+                        MaxSizeInMB = _configuration.MaxSizeInMB,
+                        DefaultMessageTimeToLive = _configuration.DefaultMessageTimeToLive
+                    });
+                _topicks.Add(topicName, new HashSet<string>());
+            }
+
+            stopwatch.Stop();
+            _logger.LogInformation($"Topick {topicName} updated for {stopwatch.ElapsedMilliseconds}ms");
+        }
+
+        public async Task UpdateQueue(string queueName)
+        {
+            _logger.LogInformation($"Updating queue {queueName}");
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+
+            if (!_queues.Contains(queueName))
+            {
+                if (!await Client.QueueExistsAsync(queueName))
+                    await Client.CreateQueueAsync(new QueueDescription(queueName)
+                    {
+                        MaxSizeInMB = _configuration.MaxSizeInMB,
+                        DefaultMessageTimeToLive = _configuration.DefaultMessageTimeToLive
+                    });
+                _queues.Add(queueName);
+            }
+
+            stopwatch.Stop();
+            _logger.LogInformation($"Queue {queueName} updated for {stopwatch.ElapsedMilliseconds}ms");
+        }
+
+        private Task<IList<RuleDescription>> GetAllRules(string topicName, string subscriptionName) =>
+            GetAll((take, skip) => Client.GetRulesAsync(topicName, subscriptionName));
+
+        private static async Task<IList<TEntity>> GetAll<TEntity>(Func<int, int, Task<IList<TEntity>>> fetch)
+        {
+            int skip = 0;
+            const int take = 100;
+
+            IList<TEntity> entitites = null;
+            do
+            {
+                var loadedRules = await fetch(take, skip);
+                if (entitites == null)
+                    entitites = loadedRules;
+                else
+                    foreach (var r in loadedRules)
+                        entitites.Add(r);
+            } while (entitites.Count % take == 0);
```

Comment:
1. skip is not incremented.
2. if even if increment skip by loadedRules.Count, make sure it does not fail when the total number of entities is 100.